agent/handlers/v4/tmdsstate_linux.go

```diff
@@ -29,8 +29,9 @@ func (s *TMDSAgentState) GetTaskMetadataWithTaskNetworkConfig(v3EndpointID strin
 	taskResponse, err := s.getTaskMetadata(v3EndpointID, false, true)
 	if err == nil {
 		if taskResponse.TaskNetworkConfig != nil && taskResponse.TaskNetworkConfig.NetworkMode == "host" {
-			hostDeviceName, err := netconfig.DefaultNetInterfaceName(networkConfigClient.NetlinkClient)
-			if err != nil {
+			hostDeviceName, netErr := netconfig.DefaultNetInterfaceName(networkConfigClient.NetlinkClient)
+			if netErr != nil {
+				err = tmdsv4.NewErrorDefaultNetworkInterfaceName(netErr.Error())
```

Comment:
ie when ecs agent can't find the given the container id, it will return this static error msg: https://github.com/aws/amazon-ecs-agent/blob/cf8c7a6b65043c550533f330b10aef6d0a342214/agent/handlers/v4/tmdsstate.go#L67-L69

Reply:
No return here?

Reply:
>IMO we can just set a static error msg as the externalReason, like unable to obtain default network interface name without original error details . And we can log the error.

Sure, I have no issues with this.

Reply:
> No return here?
The idea was for it to return in line 54. We're just setting/overriding the `err` variable with the `NewErrorDefaultNetworkInterfaceName()`.

Reply:
How about we still include the original error details in the `externalReason` but in the response back to the request we'll have a static message like `unable to obtain default network interface name on host for container: <some_container_ID>`? Thoughts @xxx0624 @amogh09 ?

Reply:
I think we can save the static message in this externalReason which is returned to the client.

ie for container metadata endpoint, agent will set the static message in the externalReason - https://github.com/aws/amazon-ecs-agent/blob/cf8c7a6b65043c550533f330b10aef6d0a342214/agent/handlers/v4/tmdsstate.go#L67-L69 and then in the handler, it will include the externalReason in the response body  - https://github.com/aws/amazon-ecs-agent/blob/cf8c7a6b65043c550533f330b10aef6d0a342214/ecs-agent/tmds/handlers/v4/handlers.go#L75

Reply:
I see, I will omit the original error encountered in `DefaultNetInterfaceName` from `externalReason` . for consistency, we might want to also do the same for the other error types -> 
https://github.com/aws/amazon-ecs-agent/blob/cf8c7a6b65043c550533f330b10aef6d0a342214/ecs-agent/tmds/handlers/fault/v1/handlers/handlers.go#L760-L769

Reply:
Yes sounds good to me.